conda_smithy/feedstock_io.py

```diff
@@ -0,0 +1,74 @@
+from contextlib import contextmanager
+import os
+import shutil
+
+
+def get_repo(path, search_parent_directories=True):
+    repo = None
+    try:
+        import git
+        repo = git.Repo(
+            path,
+            search_parent_directories=search_parent_directories
+        )
+    except ImportError:
+        pass
+    except git.InvalidGitRepositoryError:
+        pass
+
+    return repo
+
+
+def get_file_blob(repo, filename):
+    idx = repo.index
+    rel_filepath = os.path.relpath(filename, repo.working_dir)
+    blob = idx.iter_blobs(lambda _: _[1].path == rel_filepath).next()[1]
```

Comment:
Turns out the `next` method here is trouble. Particularly on Python 3. Need to use the `next` function instead.